espnet2/enh/nets/dprnn_raw.py

```diff
@@ -0,0 +1,724 @@
+        return bf_filter
+
+
+# DPRNN for beamforming filter estimation
+class BF_module(DPRNN_base):
+    def __init__(self, *args, **kwargs):
+        super(BF_module, self).__init__(*args, **kwargs)
+
+        # gated output layer
+        # self.output = nn.Sequential(
+        #     nn.Conv1d(self.feature_dim, self.feature_dim, 1), nn.Tanh() )
+        # self.output_gate = nn.Sequential(
+        #     nn.Conv1d(self.feature_dim, self.feature_dim, 1), nn.Sigmoid() )
+        self.output = nn.Sequential(
+            nn.Conv1d(self.feature_dim + 512, self.feature_dim, 1), nn.Tanh()
+        )
+        self.output_gate = nn.Sequential(
+            nn.Conv1d(self.feature_dim + 512, self.feature_dim, 1), nn.Sigmoid()
+        )
+
+    def forward(self, input, voiceP):
+        # input = input.to(device)
+        # input: (B, E, T)
+        # voiceP: (BS,topk,D)
+        batch_size, E, seq_length = input.shape
+        _, topk, D = voiceP.size()
+        assert batch_size == _
+        self.num_spk = topk
+
+        enc_feature = self.BN(input)  # (B, E, L)-->(B, N, L)
+        N = enc_feature.shape[1]
+        # split the encoder output into overlapped, longer segments
+        enc_segments, enc_rest = self.split_feature(
+            enc_feature, self.segment_size
+        )  # B, N, L, K: L is the segment_size
+        # pass to DPRNN
+        output = self.DPRNN(enc_segments)
+        output = output.view(
+            batch_size, self.feature_dim, self.segment_size, -1
+        )  # B, N, L, K
+
+        # print('output shape',output.shape)
+        # overlap-and-add of the outputs
+        output = self.merge_feature(output, enc_rest)  # B, N, L
+        output = output.unsqueeze(1).expand(-1, topk, -1, -1)  # B,topk,N,L
+        voiceP = voiceP.unsqueeze(-1).expand(
+            batch_size, topk, D, seq_length
+        )  # B,topk,D,L
+
+        # END cat mode
+        output = torch.cat((output, voiceP), dim=2).view(
+            -1, N + D, seq_length
+        )  # [B*topk,(N+D),L]
+        del voiceP
+
+        # gated output layer for filter generation
+        bf_filter = self.output(output) * self.output_gate(output)  # B*nspk, K, L
+        # self.feature_dim)  # B, nspk, L, N
+        # bf_filter = output
+
+        return bf_filter
+
+
+# base module for FaSNet
+class FaSNet_base(AbsEnhancement):
+    def __init__(
+        self,
+        fs=8000,
+        enc_dim=64,
+        feature_dim=64,
+        hidden_dim=128,
+        layer=6,
+        segment_size=250,
+        nspk=2,
+        win_len=2,
+    ):
+        """Fasnet base.
+
+        Reference:
+            "Dual-path RNN: efficient long sequence modeling for
+            time-domain single-channel speech separation", Yi Luo, Zhuo Chen
+            and Takuya Yoshioka. https://arxiv.org/abs/1910.06379
+
+        Based on https://github.com/kaituoxu/Conv-TasNet and
+              https://github.com/yluo42/TAC
+        """
+        super(FaSNet_base, self).__init__()
```

Comment:
Could you change all super(...).__init__() to super().__init__. ESPnet2 is python3 project.

Reply:
Done.